taipy/core/job/_job_manager.py

```diff
@@ -58,7 +58,9 @@ def _create(
         return job
 
     @classmethod
-    def _delete(cls, job: Job, force=False):
+    def _delete(cls, job: Union[Job, JobId], force=False):
+        if isinstance(job, str):
```

Comment:
Why str and not JobId ?
```suggestion
        if isinstance(job, JobId):
```

Reply:
Probably a copy-paste systematic error.

We use the `if isinstance(job, str)` quite a lot, and similarly for other entity types, like `if isinstance(scenario, str)` in the _ScenarioManager